third_party/move/tools/move-bytecode-utils/src/compiled_module_viewer.rs

```diff
@@ -4,11 +4,24 @@
 
 use move_binary_format::CompiledModule;
 use move_core_types::language_storage::ModuleId;
-use std::borrow::Borrow;
+use move_vm_runtime::ModuleStorage;
+use std::{borrow::Borrow, sync::Arc};
 
 pub trait CompiledModuleView {
     type Item: Borrow<CompiledModule>;
 
     // TODO: Consider using address and module name instead of module id.
     fn view_compiled_module(&self, id: &ModuleId) -> anyhow::Result<Option<Self::Item>>;
 }
+
+impl<M: ModuleStorage> CompiledModuleView for M {
```

Comment:
Is this change needed?

Reply:
No, I think we can reuse InMemoryStorage here as well.

Reply:
SG. Let's remove the unnecessary change just to make things a bit easier :D

Reply:
+1! For all Move tests, I will do a pass to simplify them as promised to @gelash already 😄 just wanted to put up the core logic out so that tests pass and things work ))

Reply:
Will post a PR with som cleanups👍